### PR TITLE
refactor refresh of wireguard keys

### PR DESCRIPTION
**Pull request by mattkasun**
## Describe your changes
key updates are handled in host updates with updateaction = UpdateKeys
## Provide testing steps
see GRA-1529

## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [x] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [ ] Netclient & Netmaker are awesome.

### CODE REVIEW

File: functions/daemon.go [hunk 1]
```diff
@@ -391,24 +393,29 @@ func unsubscribeHost(client mqtt.Client, server string) {
 }
 
 // UpdateKeys -- updates private key and returns new publickey
-func UpdateKeys(node *config.Node, host *config.Config, client mqtt.Client) error {
+func UpdateKeys(client mqtt.Client) error {
 	var err error
-	logger.Log(0, "received message to update wireguard keys for network ", node.Network)
+	logger.Log(0, "received message to update wireguard keys ")
+	host := config.Netclient()
+	pretty.Println("current keys", host.PrivateKey.String(), host.PublicKey.String())
```
Comment (0xdcarns) [inline functions/daemon.go:400]: debug log

File: functions/daemon.go [hunk 2]
```diff
@@ -391,24 +393,29 @@ func unsubscribeHost(client mqtt.Client, server string) {
...
 	host.PrivateKey, err = wgtypes.GeneratePrivateKey()
 	if err != nil {
-		logger.Log(0, "network:", node.Network, "error generating privatekey ", err.Error())
+		logger.Log(0, "error generating privatekey ", err.Error())
 		return err
 	}
+	pretty.Println("new private key", host.PrivateKey.String())
```
Comment (0xdcarns) [inline functions/daemon.go:406]: debug log

File: functions/daemon.go [hunk 3]
```diff
@@ -391,24 +393,29 @@ func unsubscribeHost(client mqtt.Client, server string) {
...
 	file := config.GetNetclientPath() + "netmaker.conf"
 	if err := wireguard.UpdatePrivateKey(file, host.PrivateKey.String()); err != nil {
-		logger.Log(0, "network:", node.Network, "error updating wireguard key ", err.Error())
+		logger.Log(0, "error updating wireguard key ", err.Error())
 		return err
 	}
 	host.PublicKey = host.PrivateKey.PublicKey()
+	pretty.Println("new public key", host.PublicKey.String())
```
Comment (0xdcarns) [inline functions/daemon.go:413]: debug log

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: functions/daemon.go (hunk 1 above)
Comment (mattkasun) [reply functions/daemon.go]: debug logs removed

**Review by theguy951357 — APPROVED** [review]
pr passes
critical list passes

**Review by Aceix — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]
pr steps pass
critical list passes

File: functions/mqhandlers.go [hunk 4]
```diff
@@ -235,6 +226,9 @@ func HostUpdate(client mqtt.Client, msg mqtt.Message) {
 		if err = PublishHostUpdate(serverName, models.Acknowledgement); err != nil {
 			logger.Log(0, "failed to response with ACK to server", serverName)
 		}
+	case models.UpdateKeys:
+		clearRetainedMsg(client, msg.Topic()) // clear message
+		UpdateKeys(client)
```
Comment (abhishek9686) [inline functions/mqhandlers.go:231]: why is it required to pass mq client as parameter to `UpdateKeys`

**Comment by mattkasun** [comment]
depends on netmaker PR -- 2199
do not merge until 2199 is merged and go.mod updated

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**